modules/multy-s3-bucket/0.2.0/s3-replica-bucket.tf

```diff
@@ -33,13 +33,102 @@ resource "aws_s3_bucket_lifecycle_configuration" "replica" {
   provider   = aws.replicaregion
   depends_on = [aws_s3_bucket_versioning.replica]
   bucket     = aws_s3_bucket.replica.id
-  rule {
-    id = "expire_non_current_version"
 
-    filter {}
+   dynamic "rule" {
+
+    for_each = aws_route53_zone.clusters
+    content {
+      id = "expire_old_vault_backups"
+
+      filter {
+        prefix = "${rule.key}/hashicorp-vault-backups/"
+      }
+
+      expiration {
+        days = 15
+      }
+
+      status = "Enabled"
+    }
+    
+  }
+
+  dynamic "rule" {
+
+    for_each = aws_route53_zone.clusters
+    content {
+      id = "expire_old_tls_backups"
```

Comment:
each lifecycle poilicy needs it's own unique id: https://docs.aws.amazon.com/AmazonS3/latest/userguide/intro-lifecycle-rules.html#intro-lifecycle-rule-id